File: Backend/AstroGame.Shared/Models/Technology.cs
using System;
using System.Collections.Generic;
using AstroGame.Shared.Models.Conditions;

namespace AstroGame.Shared.Models
{
    public abstract class Technology
    {
        public Guid Id { get; set; }

        public IEnumerable<Condition> NeededConditions { get; set; }
        public IEnumerable<Condition> ConditionFor { get; set; }
    }
}

File: Frontend/apps/astrogame_app/lib/models/base_types/technology.dart
import 'package:astrogame_app/communications/converters/condition_converter.dart';
import 'package:astrogame_app/communications/converters/guid_converter.dart';
import 'package:astrogame_app/communications/converters/technology_converter.dart';
import 'package:astrogame_app/models/conditions/condition.dart';
import 'package:flutter_guid/flutter_guid.dart';

@GuidConverter()
abstract class Technology {
  Guid id;
  List<Condition> neededConditions;
  List<Condition> conditionFor;

  Technology();

  factory Technology.fromJson(Map<String, dynamic> json) =>
      TechnologyConverter().fromJson(json);

  Map<String, dynamic> toJson() => TechnologyConverter().toJson(this);
}
